satpy/readers/oli_tirs_l1_tif.py

```diff
@@ -0,0 +1,304 @@
+
+        # Get the channel name
+        self.channel = self.get_btype(filetype_info["file_type"])
+
+        # Data can be VIS, TIR or Combined. This flag denotes what the granule contains (O, T or C respectively).
+        self.chan_selector = filename_info["data_type"]
+
+        self._obs_date = filename_info["observation_date"]
+        self._mda = mda
+
+        # Retrieve some per-band useful metadata
+        self.bsat = self._mda.band_saturation
+        self.calinfo = self._mda.band_calibration
+        self.platform_name = PLATFORMS[filename_info["spacecraft_id"]]
+
+    def get_dataset(self, key, info):
+        """Load a dataset."""
+        if self.channel != key["name"]:
+            raise ValueError(f"Requested channel {key['name']} does not match the reader channel {self.channel}")
+
+        if key["name"] in OLI_BANDLIST and self.chan_selector not in ["O", "C"]:
+            raise ValueError(f"Requested channel {key['name']} is not available in this granule")
+        if key["name"] in TIRS_BANDLIST and self.chan_selector not in ["T", "C"]:
+            raise ValueError(f"Requested channel {key['name']} is not available in this granule")
+
+        logger.debug("Reading %s.", key["name"])
+
+        data = xr.open_dataset(self.filename, engine="rasterio",
+                               chunks={"band": 1,
+                                       "y": CHUNK_SIZE,
+                                       "x": CHUNK_SIZE},
+                               mask_and_scale=False)["band_data"].squeeze()
+
+        # The fill value for Landsat is '0', for calibration simplicity convert it to np.nan
+        data = xr.where(data == 0, np.float32(np.nan), data)
+
+        attrs = data.attrs.copy()
+        # Add useful metadata to the attributes.
+        attrs["perc_cloud_cover"] = self._mda.cloud_cover
+
+        # Only OLI bands have a saturation flag
+        if key["name"] in OLI_BANDLIST:
+
+            attrs["saturated"] = self.bsat[key["name"]]
+
+        # Rename to Satpy convention
+        data = data.rename({"band": "bands"})
+
+        data.attrs.update(attrs)
+
+        # Calibrate if we're using a band rather than a QA or geometry dataset
+        if key["name"] in BANDLIST:
+            data = self.calibrate(data, key["calibration"])
+        if key["name"] in ANGLIST:
+            data.data = data.data * 0.01
+            data.attrs["units"] = "degrees"
+            data.attrs["standard_name"] = "solar_zenith_angle"
+
+        return data
+
+    def calibrate(self, data, calibration):
+        """Calibrate the data from counts into the desired units."""
+        if calibration == "counts":
+            data.attrs["standard_name"] = "counts"
+            data.attrs["units"] = "1"
+            return data
+
+        if calibration in ["radiance", "brightness_temperature"]:
+            data.attrs["standard_name"] = "toa_outgoing_radiance_per_unit_wavelength"
+            data.attrs["units"] = "W m-2 um-1 sr-1"
```

Comment:
These are provided in the yaml and made available through the info parameter (in get_dataset), would be best to use these to avoid hardcoding, right?
Same goes for the two next paragraphs...

Reply:
I thought that too, but for some reason it doesn't work. If I comment those lines out then `standard_name` and `units` are not included in the band metadata/attrs...

Reply:
Ok, what I'm saying is that the value of these should be gotten from `info`, eg:
```python
data.attrs["standard_name"] = info["standard_name"]
```
since we define them in the yaml file

Reply:
Done!